distributed/event.py

```diff
@@ -0,0 +1,251 @@
+            # No matter if someone is listening or not,
+            # we set the event to true
+            event = self._get_or_create_event(name)
+            self.scheduler.loop.add_callback(event.set)
+
+    async def event_clear(self, stream=None, name=None):
+        """Set the event with the given name to false."""
+        with log_errors():
+            name = self._normalize_name(name)
+            if not self._waiter_count[name]:
+                # No one is waiting for this
+                # and as the default flag for an event is false
+                # we can safely remove it
+                self._delete_event(name)
+
+            else:
+                # There are waiters
+                # This can happen if an event is "double-cleared"
+                # In principle, the event should be unset at this point
+                # (because if it is set, all waiters should have been
+                # notified). But to prevent race conditions
+                # do to unlucky timing, we clear anyways
+                assert name in self._events
+                event = self._events[name]
+                self.scheduler.loop.add_callback(event.clear)
+
+    async def event_is_set(self, stream=None, name=None):
+        with log_errors():
+            name = self._normalize_name(name)
+            # the default flag value is false
+            if name not in self._events:
+                return False
+
+            return self._events[name].is_set()
+
+    def _normalize_name(self, name):
+        """ Helper function to normalize an event name """
+        if isinstance(name, list):
+            name = tuple(name)
+
+        return name
+
+    def _get_or_create_event(self, name):
+        """ Helper function to return or create and return an event """
+        if name not in self._events:
+            # It seems we are the first one accessing it,
+            # so lets create a new event
+            self._events[name] = asyncio.Event()
+
+        return self._events[name]
+
+    def _delete_event(self, name):
+        """ Helper function to delete an event """
+        with suppress(KeyError):
+            del self._waiter_count[name]
+        with suppress(KeyError):
+            del self._events[name]
+
+
+class Event:
+    """ Distributed Centralized Event equivalent to asyncio.Event
+
+    An event stores a single flag, which is set to false on start.
+    The flag can be set to true (using the set() call) or back to false
+    (with the clear() call).
+    Every call to wait() blocks until the event flag is set to true.
+
+    Parameters
+    ----------
+    name: string (optional)
+        Name of the event.  Choosing the same name allows two
+        disconnected processes to coordinate an event.  If not given, a random
+        name will be generated.
+    client: Client (optional)
+        Client to use for communication with the scheduler.  If not given, the
+        default global client will be used.
```

Comment:
```suggestion
        Client to use for communication with the scheduler.  
        If not given, the default global client will be used.
```

Plus this way I get to play with multi-line suggestions :)